runtime/cudaq/builder/kernel_builder.h

```diff
@@ -491,6 +494,22 @@ class kernel_builder : public details::kernel_builder_base {
   CUDAQ_BUILDER_ADD_MEASURE(my)
   CUDAQ_BUILDER_ADD_MEASURE(mz)
 
+  /// @brief SWAP operation for swapping the quantum states of qubits.
+  /// Supports SWAP's between two qubits, as well as a multi-qubit
+  /// controlled SWAP operation.
+  void swap(QuakeValue &qubit, QuakeValue &target) {
+    std::vector<QuakeValue> empty;
+    std::vector<QuakeValue> targets{qubit, target};
+    details::swap(*opBuilder, empty, targets);
+  }
+
+  void swap(QuakeValue &&qubit, QuakeValue &&target) { swap(qubit, target); }
```

Comment:
We can replace this one and the one above it with `void swap(const QuakeValue& first, const QuakeValue& second)` which will work for both lvalue and rvalue references